src/cast_core/src/StreamStatsReporter.cpp

```diff
@@ -64,9 +64,15 @@ struct StreamStatsReporter::PhaseConfig {
             BOOST_THROW_EXCEPTION(InvalidConfigurationException("StreamStatsReporter only allows a single repeat"));
         }
 
+        std::string tenantId = phaseContext["TenantId"].to<std::string>();
         std::string streamProcessorName = phaseContext["StreamProcessorName"].to<std::string>();
         bsoncxx::builder::stream::document builder;
-        statsCommand = builder << "streams_getStats" << "" << "name" << streamProcessorName << bsoncxx::builder::stream::finalize;
+        statsCommand = builder
+            << "streams_getStats" << ""
+            << "tenantId" << tenantId
+            << "name" << streamProcessorName
+            << "processorId" << streamProcessorName
```

Comment:
`processorId` is getting set to the `streamProcessorName` here, but added to the YAML files as a separate parameter, that does not get read here?

Reply:
Either remove `processorId` from all the yaml files or add and use the following (which will default it to `streamProcessorName` , feel free to simplify if this isn't what you want):
```
        auto processorId = phaseContext["processorId"].maybe<std::string>()
                                      .value_or(streamProcessorName);
```

Reply:
Ah, thanks for pointing this out. I removed processorId from the yml files.